src/Analyzer/Passes/AggregateFunctionsArithmericOperationsPass.cpp

```diff
@@ -132,8 +133,9 @@ class AggregateFunctionsArithmericOperationsVisitor : public InDepthQueryTreeVis
             if (!function_name_if_constant_is_negative.empty() &&
                 right_argument_constant_value_literal < zeroField(right_argument_constant_value_literal))
             {
-                resolveAggregateFunctionNode(*aggregate_function_node, function_name_if_constant_is_negative);
+                lower_function_name = function_name_if_constant_is_negative;
             }
+            resolveAggregateFunctionNode(*aggregate_function_node, inner_function_arguments_nodes[0], function_name_if_constant_is_negative);
 
             auto inner_function = aggregate_function_arguments_nodes[0];
```

Comment:
I'm not sure, but probably this function should be resolved again too because it's argument type may be changed.

Reply:
I will double check it in https://github.com/ClickHouse/ClickHouse/pull/44013.